src/domain/flow-system/src/entities/shared/schedule.rs

```diff
@@ -36,12 +40,52 @@ pub struct ScheduleCronExpression {
 
 /////////////////////////////////////////////////////////////////////////////////////////
 
+#[derive(Error, Debug)]
+pub enum ScheduleError {
+    #[error(transparent)]
+    InvalidCronExptression(#[from] InvalidCronExptressionError),
+
+    #[error(transparent)]
+    CronExpressionIterationExceed(#[from] CronExpressionIterationError),
+}
+
+#[derive(Error, Debug)]
+#[error("Cron expression {expression} is invalid")]
+pub struct InvalidCronExptressionError {
+    pub expression: String,
+}
+
+#[derive(Error, Debug)]
+#[error("Cron expression {expression} iteration has been exceed")]
+pub struct CronExpressionIterationError {
+    pub expression: String,
+}
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
 impl Schedule {
-    pub fn next_activation_time(&self, now: DateTime<Utc>) -> DateTime<Utc> {
+    pub fn next_activation_time(&self, now: DateTime<Utc>) -> Result<DateTime<Utc>, ScheduleError> {
         match self {
-            Schedule::TimeDelta(td) => now + td.every,
-            Schedule::CronExpression(_) => {
-                unimplemented!()
+            Schedule::TimeDelta(td) => Ok(now + td.every),
+            Schedule::CronExpression(ce) => {
+                let schedule = match CronSchedule::from_str(&ce.cron_expression) {
+                    Err(_) => {
+                        return Err(ScheduleError::InvalidCronExptression(
+                            InvalidCronExptressionError {
+                                expression: ce.cron_expression.clone(),
+                            },
+                        ));
+                    }
+                    Ok(cron_schedule) => cron_schedule,
+                };
```

Comment:
The validation might be too late for this moment.
We want to detect it when it's configured, not when it is used.
I would expect this validation at `Schedule` construction time, and converstion into GraphQL API error, when called from there.

In addition, converting to CRON schedule each time might be expensive for every `next_activation_time` call.